### DISCUSSION TITLE
How to capture deeper spans info of libraries or Frameworks

### DISCUSSION THREAD

**Discussion by zhangjiabin1010** [open-telemetry/opentelemetry-java-instrumentation#11338] [Q&A] [answered]
I am using OTel Java Agent，but no deeper spans info of libraries.
I'm hoping to get a span similar to the one in the following picture.Is there any method or idea?

![test](https://github.com/open-telemetry/opentelemetry-java-instrumentation/assets/18608131/6c9417b5-85ad-474b-adf8-40c91005c76c)

**Comment by steverao** [comment]
> I am using otlp as receiver.

Don't you use OTel Java Agent to collect observable data?

**Reply by zhangjiabin1010** [reply]
I am using OTel Java Agent， then sends the data to collector。I observed that the current agent can only capture relatively shallow spans.

**Reply by zhangjiabin1010** [reply]
For example, a database query can only capture the executed sql, but not the methods executed in postgresql-jdbc-driver.

**Reply by laurit** [reply, marked as answer]
Perhaps what you really want is profiling?
I have the feeling that having too fine grained spans isn't generally too useful. If you need these spans then the agent offers an extension mechanism that you can use to write the instrumentations that suite your needs. If you believe that such instrumentations could be useful for wider audience then these may be added to this or the contrib repo or you can host them on your own.

**Reply by zhangjiabin1010** [reply]
OK, I'll try to implement it myself
